src/Compilers/CSharp/Portable/Symbols/MethodSymbolExtensions.cs

```diff
@@ -300,5 +301,18 @@ public static bool IsGenericTaskReturningAsync(this MethodSymbol method, CSharpC
             return method.IsAsync
                 && method.ReturnType.IsGenericTaskType(compilation);
         }
+
+        internal static CSharpSyntaxNode ExtractReturnTypeSyntax(this MethodSymbol method)
+        {
+            foreach (var reference in method.DeclaringSyntaxReferences)
+            {
+                var methodDeclaration = reference.GetSyntax() as MethodDeclarationSyntax;
+                if ((object)methodDeclaration != null)
+                {
+                    return methodDeclaration.ReturnType;
+                }
+            }
+            return CSharpSyntaxTree.Dummy.GetRoot() as CSharpSyntaxNode;
```

Comment:
>CSharpSyntaxTree.Dummy.GetRoot() as CSharpSyntaxNode [](start = 19, length = 52)

```as``` cast should not be used when we don't expect it to fail. If we do expect it to fail, we should handle null result, which I do not see at the call site.